### PR TITLE
split up CUDA-suffixed dependencies in dependencies.yaml

### PR DESCRIPTION
**Pull request by jameslamb**
## Description

Contributes to https://github.com/rapidsai/build-planning/issues/31

In short, RAPIDS DLFW builds want to produce wheels with unsuffixed dependencies, e.g. `cudf` depending on `rmm`, not `rmm-cu12`.

This PR is part of a series across all of RAPIDS to try to support that type of build by setting up CUDA-suffixed and CUDA-unsuffixed dependency lists in `dependencies.yaml`.

For more details, see:
* https://github.com/rapidsai/build-planning/issues/31#issuecomment-2245815818
* https://github.com/rapidsai/cudf/pull/16183

### Why target 24.08?

This is targeting 24.08 because:

1. it should be very low-risk
2. getting these changes into 24.08 prevents the need to carry around patches for every library in DLFW builds using RAPIDS 24.08

### CODE REVIEW

**Comment by jameslamb** [comment]
I've updated this based on the suggestions from https://github.com/rapidsai/cudf/pull/16183.

Ran the following to check `update-version.sh`.

```shell
git fetch upstream --tags
ci/release/update-version.sh '24.10.00'

git grep -E '24\.8|24\.08|0\.39'
```

No changes to that script were required.

File: dependencies.yaml [hunk 1]
```diff
@@ -562,7 +578,7 @@ dependencies:
     common:
       - output_types: conda
         packages:
-          - &ucx_conda_run ucx>=1.15.0
```
Comment (vyasr) [inline dependencies.yaml:565]: I guess this anchor was just unused previously?

File: dependencies.yaml [hunk 2]
```diff
@@ -529,18 +537,22 @@ dependencies:
...
+              cuda_suffixed: "true"
             packages:
               - distributed-ucxx-cu11==0.39.*,>=0.0.0a0
-          - {matrix: null, packages: [*distributed_ucxx_conda]}
+          - {matrix: null, packages: [*distributed_ucxx_unsuffixed]}
   depends_on_ucx_build:
     common:
       - output_types: conda
         packages:
-          - &ucx_conda_build ucx==1.15.0
```
Comment (vyasr) [inline dependencies.yaml:543]: Previously unused anchor?

**Review by vyasr — APPROVED** [review]

File: dependencies.yaml (hunk 1 above)
Comment (jameslamb) [reply dependencies.yaml:565]: yeah exactly. Maybe I should have left that out of the diff, but figured it was low-risk, non-controversial, and related-enough to include here.

File: dependencies.yaml (hunk 2 above)
Comment (jameslamb) [reply dependencies.yaml:543]: yep: https://github.com/rapidsai/raft/pull/2388#discussion_r1690518350

**Review by bdice — APPROVED** [review]

**Comment by jameslamb** [comment]
queuing this up to auto-merge whenever that last test job is picked up. I'll check back in a bit to see if anything needs to be manually restarted.

**Pull request merged**